File: Data Structures & Algorithms/majority-element/submission-0.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {

        int numOfEle = nums.size();
        unordered_map<int,int> freq;

        for(auto ele : nums){
            freq[ele]++;
        }

        for(auto pair : freq){
            if(pair.second > floor(numOfEle/2)) return pair.first;
        }

        
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        int numOfEle = nums.length;

        int[] result = new int[numOfEle];

        result[0] = 1;

        for(int i = 1; i < numOfEle; i++){
            result[i] = result[i -1] * nums[i - 1];
        }

        int postFix = 1;

        for(int i = numOfEle -1; i >= 0; i--){
            result[i] *= postFix;
            postFix *= nums[i];
        }

        return result;
    }
}  
